### PR TITLE
69 implementing data add

### LINKED ISSUE #75: Implement Data.add()

### PR DESCRIPTION
**Pull request by parthk279**

### CODE REVIEW

File: code/Data.py [hunk 1]
```diff
@@ -14,9 +20,17 @@ def __init__(self, src):
...
-        self.cols = xs
+    
+    def add(self, xs: Row):
+
+        if not self.cols:
+            self.cols = Cols(xs) 
+        else:
+            row = xs if type(xs)==Row else Row(xs)    
+            self.rows.append(row)
+            for todo in [self.cols.x, self.cols.y]:
```
Comment (Nikhil1912) [inline code/Data.py:31]: Are you sure about this? Can you explain or add comments on what this part is supposed to do?

File: code/Data.py [hunk 2]
```diff
@@ -14,9 +20,17 @@ def __init__(self, src):
...
-    def add(self, xs):
-        # Filler code this is TODO
-        self.cols = xs
+    
+    def add(self, xs: Row):
+
+        if not self.cols:
+            self.cols = Cols(xs) 
+        else:
+            row = xs if type(xs)==Row else Row(xs)
```
Comment (parthk279) [inline code/Data.py:29]: Checks if xs is of Row type, if not convert it to Row

File: code/Data.py (hunk 1 above)
Comment (parthk279) [reply code/Data.py:31]: Runs a loop through both x and y(dependant and independent columns)

File: code/Data.py [hunk 3]
```diff
@@ -14,9 +20,17 @@ def __init__(self, src):
...
+    def add(self, xs: Row):
+
+        if not self.cols:
+            self.cols = Cols(xs) 
+        else:
+            row = xs if type(xs)==Row else Row(xs)    
+            self.rows.append(row)
+            for todo in [self.cols.x, self.cols.y]:
+                for col in todo:
+                    col.add(row["cells"][col.c])
```
Comment (parthk279) [inline code/Data.py]: Adds data to col, although I might have used an incorrect variable for column position. Which I'll amend

**Pull request merged**